http-service/src/main/java/net/runelite/http/service/hiscore/HiscoreService.java

```diff
@@ -39,34 +40,48 @@
 import org.apache.commons.csv.CSVRecord;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
-import org.springframework.web.bind.annotation.PathVariable;
-import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.bind.annotation.RequestParam;
-import org.springframework.web.bind.annotation.RestController;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.WebDataBinder;
+import org.springframework.web.bind.annotation.*;
 
 @RestController
 @RequestMapping("/hiscore")
 public class HiscoreService
 {
 	private static final Logger logger = LoggerFactory.getLogger(HiscoreService.class);
 
-	private static final HttpUrl RUNESCAPE_HISCORE_SERVICE = HttpUrl.parse("http://services.runescape.com/m=hiscore_oldschool/index_lite.ws");
-
-	private HttpUrl url = RUNESCAPE_HISCORE_SERVICE;
+	HiscoreResultBuilder lookupUsername(String username, HiscoreEndpoint endpoint) throws IOException
+	{
+		return lookupUsername(username, HttpUrl.parse(endpoint.hiscoreUrl()));
+	}
 
-	private HiscoreResultBuilder lookupUsername(String username) throws IOException
+	HiscoreResultBuilder lookupUsername(String username, HttpUrl hiscoreUrl) throws IOException
 	{
-		HttpUrl hiscoreUrl = url.newBuilder()
-			.addQueryParameter("player", username)
-			.build();
+		HttpUrl url = hiscoreUrl.newBuilder()
+				.addQueryParameter("player", username)
+				.build();
 
-		logger.info("Built URL {}", hiscoreUrl);
+		logger.info("Built URL {}", url);
 
 		Request okrequest = new Request.Builder()
-			.url(hiscoreUrl)
-			.build();
+				.url(url)
+				.build();
 
 		Response okresponse = RuneliteAPI.CLIENT.newCall(okrequest).execute();
+
+		if (!okresponse.isSuccessful())
+		{
+			switch (HttpStatus.valueOf(okresponse.code()))
+			{
+				case NOT_FOUND:
+					throw new NotFoundException();
+				case GATEWAY_TIMEOUT:
```

Comment:
I don't see why this needs to be special cased as it is an error from Jagex's servers